src/js/components/FileInput/FileInput.js

```diff
@@ -396,16 +397,46 @@ const FileInput = forwardRef(
           onChange={(event) => {
             event.persist();
             const fileList = event.target.files;
+            let filesToAdd;
+            if (!files.length) {
+              if (multiple && max && fileList.length >= max) {
+                const newFileList = [];
+                for (let i = 0; i < max; i += 1) {
+                  newFileList.push(fileList[i]);
+                }
+                filesToAdd = newFileList;
+              } else {
+                filesToAdd = fileList;
+              }
+            } else if (
+              multiple &&
+              max &&
+              fileList.length >= max - files.length
+            ) {
+              const newFileList = [];
+              for (let i = 0; i < max - files.length; i += 1) {
+                newFileList.push(fileList[i]);
+              }
+              // if the maximum value is exceeded then
+              // remove files from the event also.
+              if (!newFileList.length) {
+                /* eslint-disable-next-line no-param-reassign */
```

Comment:
@ericsoderberghp are we good with this eslint-disable?

Reply:
I don't think we should touch the `event.target.value`. The way this component works, we're guiding callers to use the `files` sent to `onChange` instead of `event.target.value` because the `event.target.value` for a file input is read-only.